### PR TITLE
Test of listening ports, closed port listen

### PR DESCRIPTION
**Pull request by rdkdd**
Enhancement of https://github.com/spiceqa/tp-spice/pull/52

### CODE REVIEW

**Comment by Andrei-Stepanov** [comment]
Does this one replace previous active PR : https://github.com/spiceqa/tp-spice/pull/52  ?

If yes, then please close #52

**Comment by rdkdd** [comment]
ok I have closed it.

File: spice/tests/vm_listen.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+from virttest import env_process
+from virttest import virt_vm
+
+from avocado.utils import process
+
+logger = logging.getLogger(__name__)
+
+
+@error.context_aware
```
Comment (Andrei-Stepanov) [inline spice/tests/vm_listen.py]: do you know for what this decorator?  I don't)

File: spice/tests/vm_listen.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    ----------
+    vt_test : avocado.core.plugins.vt.VirtTest
+        QEMU test object.
+    test_params : virttest.utils_params.Params
+        Dictionary with the test parameters.
+    env : virttest.utils_env.Env
+        Dictionary with test environment.
+    """
+    if test_params['start_vm'] == "no":
+        test_params['start_vm'] = "yes"
```
Comment (Andrei-Stepanov) [inline spice/tests/vm_listen.py]: We do not modify "Input" (test params).
It is bad approach.

File: spice/tests/vm_listen.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        if test_params.has_key('spice_port_closed'):
+            cmd = "nc -l %s" % test_params['spice_port_closed']
+            nc_process = process.SubProcess(cmd)
+            nc_process_pid = nc_process.start()
+            test_params['spice_port'] = test_params['spice_port_closed']
```
Comment (Andrei-Stepanov) [inline spice/tests/vm_listen.py]: Same here.

File: spice/tests/vm_listen.py (hunk 2 above)
Comment (Andrei-Stepanov) [inline spice/tests/vm_listen.py]: Better change config accordingly

File: spice/tests/vm_listen.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        error.context("Start guest VM with invalid parameters.")
+        try:
+            env_process.preprocess_vm(
+                vt_test, test_params, env, test_params["main_vm"])
+        except virt_vm.VMCreateError, emsg:
+            error_s = test_params['error_msg']
+            if '%s,%s' in error_s:
+                s_port = env.get_vm(test_params['main_vm']).spice_port
```
Comment (Andrei-Stepanov) [inline spice/tests/vm_listen.py]: Please assign a variable and replace :
 test_params['main_vm']
with this variable

File: spice/tests/vm_listen.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+                error_s = error_s % (test_params['spice_addr'], s_port)
+            if error_s in emsg.output and emsg.status == 1:
+                logging.info("Guest terminated as expected: %s" % emsg.output)
+                return
+            else:
+                raise error.TestFail("Guest creation failed, bad error message:"
+                " %s and/or exit status: %s" % (emsg.output, emsg.status))
+        finally:
+            try:
+                process.safe_kill(nc_process_pid, signal.SIGKILL)
```
Comment (Andrei-Stepanov) [inline spice/tests/vm_listen.py]: I think it will be killed automatically

File: spice/tests/vm_listen.py (hunk 2 above)
Comment (rdkdd) [inline spice/tests/vm_listen.py]: Well I must change config in the runtime. How to do this? Check your email if you want to know more.

Comment (Andrei-Stepanov) [inline spice/tests/vm_listen.py]: Ok. please add comment with link:

https://www.redhat.com/archives/avocado-devel/2017-January/msg00012.html

File: spice/tests/vm_listen.py (hunk 5 above)
Comment (rdkdd) [inline spice/tests/vm_listen.py]: no it wont it is independent process of nc -l $port

Comment (Andrei-Stepanov) [inline spice/tests/vm_listen.py]: I doubt.

You run it using:

class SubProcess(object):

    """
    Run a subprocess in the background, collecting stdout/stderr streams.
    """
./avocado/utils/process.py

Comment (Andrei-Stepanov) [inline spice/tests/vm_listen.py]: subprocess depends on parent process.
Parent process is your test.

**Pull request merged**